### PR TITLE
Add debug symbols to build + some improved logging

### PR DESCRIPTION
**Pull request by kantai**
This adds debug symbols to our release profile, plus improves some debug logging.

Also, while working with `Cargo.toml`, added @lgalabru to author list.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,19 @@
 [package]
 name = "blockstack-core"
 version = "0.0.1"
-authors = ["Jude Nelson <jude@blockstack.com>", "Aaron Blankstein <aaron@blockstack.com>"]
+authors = [ "Jude Nelson <jude@blockstack.com>",
+            "Aaron Blankstein <aaron@blockstack.com>",
+            "Ludo Galabru <ludovic@blockstack.com>" ]
```
Comment (lgalabru) [inline Cargo.toml:6]: 🥰

File: Cargo.toml [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 license = "GPLv3"
 homepage = "https://github.com/blockstack/blockstack-core"
 repository = "https://github.com/blockstack/blockstack-core"
 description = "Reference implementation of Blockstack Core"
 keywords = [ "bitcoin", "crypto", "blockstack", "decentralized", "dapps", "blockchain" ]
 readme = "README.md"
 
+[profile.release]
+debug = true
```
Comment (lgalabru) [inline Cargo.toml:15]: I think we should also add this config to the `testnet/src/Cargo.toml` (where the fields authors/version should be stripped)

Comment (jcnelson) [reply Cargo.toml:15]: Seconded

Comment (kantai) [reply Cargo.toml:15]: The top-level Cargo.toml overrides any profile settings in the other workspace Cargo.toml files -- so if we add that to the other Cargo.tomls, all we'll get is warnings about how they are ignored.

**Comment by kantai** [comment]
Okay -- so I think that something with the `musl` target is making it pretty hard to get useful debug symbols in most of the threads (for whatever reason, the only thread getting symbols is the p2p thread) -- so what I think we can do is add another Docker target `stacks-blockchain-stretch` based on a `stretch` image, and then setup our servers to run that so that we have something that's got a bit more debugability.

Thoughts, @jcnelson and @lgalabru ?

**Comment by lgalabru** [comment]
Because of the feature flag, the command for building the `stacks-node` looks like this:
```
RUN cd testnet/stacks-node && cargo build --features "monitoring_prom" --release --target x86_64-unknown-linux-musl
```

I understand that the top level `Cargo.toml` is supposed to override the settings, but in our case, our entry point is not the top level `Cargo.toml`. With the current setup, I would not be surprise if the stacks lib was compiled with symbols, and the stacks node not.

**Comment by jcnelson** [comment]
> With the current setup, I would not be surprised if the stacks lib was compiled with symbols, and the stacks node not.

This would be consistent with what we're seeing in GDB.

**Comment by kantai** [comment]
> I understand that the top level Cargo.toml is supposed to override the settings, but in our case, our entry point is not the top level Cargo.toml. With the current setup, I would not be surprised if the stacks lib was compiled with symbols, and the stacks node not.

I initially thought this was the case, but the compiler at least _claims_ to be adding debuginfo to these builds. In the Docker build output you can see:

```
...
   Compiling stacks-node v0.1.0 (/src/testnet/stacks-node)
    Finished release [optimized + debuginfo] target(s) in 8m 35s
...
```

And, when I tested this with just a native build, my lldb has no problem reading the symbols from the output.

I also tested this with a "stretch" docker image, running lldb in it with the release built `stacks-node` binary, and it has the correct backtrace symbols:

```Dockerfile
FROM rust:stretch as build

WORKDIR /src

COPY . .

RUN cd testnet/stacks-node && cargo build --release --features "monitoring_prom"
RUN cd testnet/bitcoin-neon-controller && cargo build --release

RUN cp target/release/stacks-node /bin/
RUN cp target/release/bitcoin-neon-controller /bin/

CMD ["stacks-node", "argon"]
```

```
# lldb stacks-node
(lldb) target create "stacks-node"
Current executable set to 'stacks-node' (x86_64).
(lldb) r argon
Process 378 launched: '/bin/stacks-node' (x86_64)
INFO [1595352523.538] [testnet/stacks-node/src/run_loop/neon.rs:98] [ThreadId(1)] Follower node: starting up
ERROR [1595352523.545] [src/chainstate/stacks/index/storage.rs:1175] [ThreadId(1)] Not found (no file is open)
INFO [1595352523.545] [src/chainstate/stacks/index/marf.rs:751] [ThreadId(1)] First-ever block 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206 in /tmp/stacks-testnet-1d0d19b336ad9ab2/burnchain/db/bitcoin/regtest/sortition.db/marf
Process 378 stopped
* thread #1: tid = 378, 0x00007ffff77c04bd libpthread.so.0`__libc_recv + 29, name = 'stacks-node', stop reason = signal SIGSTOP
    frame #0: 0x00007ffff77c04bd libpthread.so.0`__libc_recv + 29
libpthread.so.0`__libc_recv:
->  0x7ffff77c04bd <+29>: cmpq   $-0x1000, %rax            ; imm = 0xFFFFFFFFFFFFF000 
    0x7ffff77c04c3 <+35>: ja     0x7ffff77c04d0            ; <+48>
    0x7ffff77c04c5 <+37>: retq   
    0x7ffff77c04c6 <+38>: nopw   %cs:(%rax,%rax)
(lldb) bt
* thread #1: tid = 378, 0x00007ffff77c04bd libpthread.so.0`__libc_recv + 29, name = 'stacks-node', stop reason = signal SIGSTOP
  * frame #0: 0x00007ffff77c04bd libpthread.so.0`__libc_recv + 29
    frame #1: 0x000055555620bad4 stacks-node`_$LT$std..net..tcp..TcpStream$u20$as$u20$std..io..Read$GT$::read::h5d280ba1d5cd9da7 + 20 at net.rs:212
    frame #2: 0x0000555555cd7601 stacks-node`std::io::Read::read_exact::h5940ef7c13010d4a + 113 at mod.rs:787
    frame #3: 0x0000555555dc02f4 stacks-node`_$LT$blockstack_lib..deps..bitcoin..network..message..RawNetworkMessage$u20$as$u20$blockstack_lib..deps..bitcoin..network..encodable..ConsensusDecodable$LT$D$GT$$GT$::consensus_decode::heee15cdddcf25125 + 1892 at impls.rs:45
    frame #4: 0x0000555555a6e95f stacks-node`blockstack_lib::burnchains::bitcoin::network::_$LT$impl$u20$blockstack_lib..burnchains..bitcoin..indexer..BitcoinIndexer$GT$::recv_message::he85faa2f1e3a317e + 111 at network.rs:76
    frame #5: 0x0000555555a64fcb stacks-node`blockstack_lib::burnchains::bitcoin::indexer::BitcoinIndexer::peer_communicate::h2c5666be6b85a678 + 1467 at indexer.rs:410
    frame #6: 0x0000555555a6b8a3 stacks-node`_$LT$blockstack_lib..burnchains..bitcoin..indexer..BitcoinIndexer$u20$as$u20$blockstack_lib..burnchains..indexer..BurnchainIndexer$GT$::find_chain_reorg::h00aab575d37d2216 + 2659 at spv.rs:237
    frame #7: 0x000055555567ebc4 stacks-node`blockstack_lib::burnchains::burnchain::_$LT$impl$u20$blockstack_lib..burnchains..Burnchain$GT$::sync_with_indexer::hecb13a3d1bd46ea7 + 5556 at burnchain.rs:618
    frame #8: 0x000055555579bc61 stacks-node`_ZN11stacks_node10burnchains26bitcoin_regtest_controller24BitcoinRegtestController14receive_blocks17haebbcaf27b609a48E.llvm.7695957293974606313 + 1873 at bitcoin_regtest_controller.rs:150
    frame #9: 0x00005555556bba94 stacks-node`stacks_node::run_loop::neon::RunLoop::start::he05651539bf5619c + 3524 at bitcoin_regtest_controller.rs:571
    frame #10: 0x000055555574297d stacks-node`stacks_node::main::he81da2b9703eb65c + 2317 at main.rs:92
    frame #11: 0x000055555566de83 stacks-node`std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::h5975e0e3b5a97efc + 3 at rt.rs:67
    frame #12: 0x0000555556211df8 stacks-node`std::rt::lang_start_internal::h464df2bbf46c7e7c + 680 at rt.rs:52
    frame #13: 0x00005555557435c8 stacks-node`main + 40
    frame #14: 0x00007ffff721a2e1 libc.so.6`__libc_start_main + 241
    frame #15: 0x00005555556570ca stacks-node`_start + 42
```

**Comment by jcnelson** [comment]
Bizarre.  If we can't figure out why the symbols aren't showing up in the static musl build, then I'd be fine with doing a glibc build that we can debug.  In my experience with musl, the only big "gotcha" I encountered with it that I didn't see in glibc is that musl has a different (smaller) thread stack size, so you can get a Rust program to segfault from a stack overflow with a musl build that won't be reproducible with a glibc build.

**Comment by jcnelson** [comment]
To make things _even weirder_, when I build and run the `stacks-node` binary on my local machine (which is `x86_64-unknown-linux-musl`), I get all the debug symbols.  I'm running musl libc.

**Comment by jcnelson** [comment]
Ooooh, I think I see what's wrong -- I think you need to `apt-get install musl-dev` as well as `musl-tools` to get the debug symbols.

**Comment by kantai** [comment]
> To make things even weirder, when I build and run the stacks-node binary on my local machine (which is x86_64-unknown-linux-musl), I get all the debug symbols. I'm running musl libc. Moreover, I can use rust-gdb to get them.

Very strange indeed -- that makes me want to look at my other suspicion, which is that the way we're cross-compiling is breaking the symbols table.

**Comment by jcnelson** [comment]
Update:  My target is `x86_64-alpine-linux-musl`, not `x86_64-unknown-linux-musl`.  Not sure if it makes a difference.

**Comment by kantai** [comment]
Well, something worked. Updating the build image in the Dockerfile to be `rust:alpine` so that it doesn't need to cross-compile, plus using lldb rather than gdb produced the expected results.

**Pull request merged**